docs/v1/P4Runtime-Spec.mdk

```diff
@@ -6269,6 +6279,356 @@ field [@ProtoAny]. At the moment, there is not any mechanism to extend the
 `p4.v1.TableEntry` message based on the value of architecture-specific table
 properties, but we may include on in future versions of the API.
 
+# GenericTable { #sec-generic-table}
+
+## GenericTable p4info { #sec-p4info-generic-table}
+
+`GenericTable` messages are used to map non-PSA P4-externs or non-P4 target-
+specific "fixed" features to their implementation in a generic way. The same can be achieved
+via `Extern`, but GenericTable provides a structured way in which every feature is
+represented as a set of match-fields and data-fields. The data consists of unions
+which are similar to actions. One use of GenericTable in a server backend is
+with TDI [Table Driven Interface](https://github.com/p4lang/tdi), but targets
+can use Generic table to map to their own specific feature implementations as
+well.
+It defines the following fields
+
+* `generic_table_type_id`, a 8-bit unsigned integer which uniquely identifies the
+  generic_table_type in the context of the architecture.
+  This value is in the range of `[0x00, 0xff]`. 
+  The ID in the preamble is created as `0x18` `generic_table_type_id` `16 bits of
+  generated ID`.
+  Note that this value does not need
+  to be unique across all architectures from all organizations, since at any
+  given time every device managed by a P4Runtime server maps to a single P4Info
+  message and a single architecture.
+
+* `generic_table_type_name`, which specifies the fully-qualified P4 name of the
+  generic_table in case of P4 entities. In cases of non-P4 entities, it is a
+  name which uniquely identifies this entity in the entire space including P4
+  objects
+
+* `generic_table_category_type`, please check section [GenericTable categories](#sec-generic-table-categories) for more details
+
+
```

Comment:
Is it the plan to have a separate GenericTable and GenericTableInstance message?  If so, as the current P4Info proto file definition in this PR says, then this documentation should say that, and separate the list of fields above as fields of GenericTable, from the list of fields below as fields of GenericTableInstance.

Either that, or leave this as is and get rid of the idea of grouping GenericTableInstance's with the same type into one GenericTable message, and just have every GenericTable describe one instance, and get rid of message GenericTableInstance.

Reply:
Done. Added text in the spec